hooks/shotgun_filters.py

```diff
@@ -0,0 +1,222 @@
+
+        elif sg_location.entity_type == "Task":
+
+            # tasks are usually associated via a task field rather than via a link field
+            if entity_type == "Note":
+                link_filters.append(["tasks", "in", [sg_location.entity_dict]])
+
+            elif entity_type == "Version":
+                link_filters.append(["sg_task", "is", sg_location.entity_dict])
+
+            elif entity_type in ["PublishedFile", "TankPublishedFile"]:
+                link_filters.append(["task", "is", sg_location.entity_dict])
+
+            elif entity_type == "Task":
+                link_filters.append(["sibling_tasks", "is", sg_location.entity_dict])
+
+            else:
+                link_filters.append(["entity", "is", sg_location.entity_dict])
+
+        elif sg_location.entity_type == "Project":
+
+            # tasks are usually associated via a task field rather than via a link field
+            if entity_type == "Note":
+                link_filters.append(["project", "is", sg_location.entity_dict])
+
+            elif entity_type == "Version":
+                link_filters.append(["project", "is", sg_location.entity_dict])
+
+            elif entity_type in ["PublishedFile", "TankPublishedFile"]:
+                link_filters.append(["project", "is", sg_location.entity_dict])
+
+            elif entity_type == "Task":
+                # my tasks tab on project
+                if context_user is None:
+                    raise sgtk.TankError(
+                        "Use of the My Tasks tab is not supported when a current Shotgun user "
+                        "cannot be determined. This is most often the case when a script key "
+                        "is used for authentication rather than a user name and password."
+                    )
+
+                link_filters.append(["task_assignees", "in", [context_user]])
+                link_filters.append(["sg_status_list", "is_not", "fin"])
+                link_filters.append(["project", "is", sg_location.entity_dict])
+
+            else:
+                link_filters.append(["entity", "is", sg_location.entity_dict])
+
+        elif sg_location.entity_type == "Version":
+
+            if entity_type == "Note":
+                link_filters.append(["note_links", "in", [sg_location.entity_dict]])
+
+            elif entity_type in ["PublishedFile", "TankPublishedFile"]:
+                link_filters.append(["version", "is", sg_location.entity_dict])
+
+            else:
+                link_filters.append(["entity", "is", sg_location.entity_dict])
+
+        else:
+            if entity_type == "Note":
+                link_filters.append(["note_links", "in", [sg_location.entity_dict]])
+            else:
+                link_filters.append(["entity", "is", sg_location.entity_dict])
+
+        self.logger.debug(
+            "%s Resolved %s into the following sg query:\n%s"
+            % (self, sg_location, pprint.pformat(link_filters))
+        )
```

Comment:
```suggestion
        self.logger.debug(
            "%s Resolved %s into the following sg query:\n%s",
            (self, sg_location, pprint.pformat(link_filters))
        )
```

Reply:
This suggested change is causing an error on Windows:
```
TypeError: not enough arguments for format string
```

So I've left it as is for now.

Reply:
Okay. That's weird, but of little/no consequence. Let's just leave it as is.